src/base/lock_free.c

```diff
@@ -893,6 +893,20 @@ lf_freelist_retire (LF_TRAN_ENTRY * tran_entry, LF_FREELIST * freelist, void *en
 	}
     }
 
+  /* free entry if alloc_cnt is greater than max_alloc_cnt */
+  if (freelist->alloc_cnt > edesc->max_alloc_cnt)
```

Comment:
it looks right "freelist->alloc_cnt >= edesc->max_alloc_cnt" ?

Reply:
It would be better to handle the problem of overflow occurring when freelist->alloc_cnt exceeds max int in another issue. '>' is correct because it is free for locks exceeding 'lock escalation'.